src/theory/builtin/proof_checker.cpp

```diff
@@ -244,6 +254,13 @@ Node BuiltinProofRuleChecker::checkInternal(PfRule id,
     Node res = applyRewrite(args[0]);
     return args[0].eqNode(res);
   }
+  else if (id == PfRule::THEORY_REWRITE)
+  {
+    Assert(children.empty());
+    Assert(args.size() == 2);
+    Node res = applyTheoryRewrite(args[0], args[1].getConst<bool>());
```

Comment:
You'll need to convert to Skolem and then back to witness form here, see for example the rules in this block: https://github.com/ajreynol/CVC4/blob/stringsPf/src/theory/strings/proof_checker.cpp#L327.